src/util.rs

```diff
@@ -126,3 +126,17 @@ pub fn short_file_path<'a>(target: &'a str, current_dir: &str) -> &'a str {
         .and_then(|p| p.to_str())
         .unwrap_or(target)
 }
+
+pub fn meta_for_session(session: String, client: Option<String>) -> EditorMeta {
```

Comment:
making this a free function is good but I think it should stay in `context.rs` which is a better name than the low-entropy `util.rs`